cmake/module/FlagsSummary.cmake

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2024-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+include_guard(GLOBAL)
+
+function(indent_message header content indent_num)
+  if(indent_num GREATER 0)
+    string(REPEAT " " ${indent_num} indentation)
+    string(REPEAT "." ${indent_num} tail)
+    string(REGEX REPLACE "${tail}$" "" header "${header}")
+  endif()
+  message("${indentation}${header} ${content}")
```

Comment:
I noticed that we're not using the `message(STATUS ` construct in this file, was that deliberate?

Reply:
From the `message` command [docs](https://cmake.org/cmake/help/latest/command/message.html):
> The CMake command-line tool displays `STATUS` ... messages on stdout with the message preceded by two hyphens and a space.

I'm afraid that two hyphens in every line could clutter the summary.